modules/cli/src/main/scala/scala/cli/config/ConfigDb.scala

```diff
@@ -0,0 +1,163 @@
+package scala.cli.config
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import com.github.plokhotnyuk.jsoniter_scala.macros._
+import coursier.parse.RawJson
+
+import java.nio.file.attribute.PosixFilePermission
+
+import scala.build.Directories
+import scala.build.errors.BuildException
+import scala.collection.immutable.ListMap
+
+final class ConfigDb private (
```

Comment:
wondering if adding a scaladoc here wouldn't be helpful long term (applies to most newly created source files in this PR as well)